sdk/core/core-http/src/policies/compressPolicy.browser.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/*
+ * NOTE: When moving this file, please update "browser" section in package.json
+ */
+import { BaseRequestPolicy, RequestPolicy, RequestPolicyOptions } from './requestPolicy';
+import { WebResource } from '../webResource';
+import { HttpOperationResponse } from '../httpOperationResponse';
+
+const compressNotSupportedInBrowser = new Error("CompressPolicy is not supported in browser environment");
+
+export interface CompressOptions {
+  enable: boolean;
+}
+
+export const DefaultCompressOptions: CompressOptions = {
+  enable: true
+}
+
+export function compressPolicy(_compressOptions?: CompressOptions) {
+  return {
+    create: (_nextPolicy: RequestPolicy, _options: RequestPolicyOptions) => {
+      throw compressNotSupportedInBrowser;
+    }
+  };
+}
+
+export class CompressPolicy extends BaseRequestPolicy {
```

Comment:
do we need this class if the factory throws on its own?

Reply:
This matches the api in the NodeJS version.